packages/calcite-components/src/components/dropdown-item/dropdown-item.scss

```diff
@@ -1,195 +1,180 @@
+  --calcite-internal-dropdown-item-icon-space-x-lg: theme("margin.2");
+  --calcite-internal-dropdown-item-icon-space-x-sm: theme("margin.1");
+  --calcite-internal-dropdown-item-icon-space-x-start: theme("spacing.1");
+  --calcite-internal-dropdown-item-line-height: 1rem /* 16px */;
+  --calcite-internal-dropdown-item-space-x-end: theme("padding.2");
+  --calcite-internal-dropdown-item-space-x-start: theme("padding.6");
+  --calcite-internal-dropdown-item-space-y: 0.25rem /* 4px */;
+
+  // none-selection mode
+  &.container--none-selection {
+    --calcite-internal-dropdown-item-space-x-start: theme("padding.1");
   }
-
-  .container--link .dropdown-link {
-    @apply text-color-1;
+}
+.container--m {
+  --calcite-internal-dropdown-item-font-size: var(--calcite-font-size--1);
+  --calcite-internal-dropdown-item-icon-space-x-lg: theme("margin.3");
+  --calcite-internal-dropdown-item-icon-space-x-sm: theme("margin.1");
+  --calcite-internal-dropdown-item-icon-space-x-start: theme("spacing.2");
+  --calcite-internal-dropdown-item-line-height: 1rem /* 16px */;
+  --calcite-internal-dropdown-item-space-x-end: theme("padding.3");
+  --calcite-internal-dropdown-item-space-x-start: theme("padding.8");
+  --calcite-internal-dropdown-item-space-y: 0.5rem /* 8px */;
+
+  &.container--none-selection {
+    --calcite-internal-dropdown-item-space-x-start: theme("padding.2");
   }
 }
-
-:host(:active:not([disabled])) .container {
-  @apply bg-foreground-3;
+.container--l {
+  --calcite-internal-dropdown-item-font-size: var(--calcite-font-size-0);
+  --calcite-internal-dropdown-item-icon-space-x-lg: theme("margin.4");
+  --calcite-internal-dropdown-item-icon-space-x-sm: theme("margin.1");
+  --calcite-internal-dropdown-item-icon-space-x-start: theme("spacing.3");
+  --calcite-internal-dropdown-item-line-height: 1.25rem /* 20px */;
+  --calcite-internal-dropdown-item-space-x-end: theme("padding.4");
+  --calcite-internal-dropdown-item-space-x-start: theme("padding.10");
+  --calcite-internal-dropdown-item-space-y: 0.625rem /* 10px */;
+  &.container--none-selection {
+    --calcite-internal-dropdown-item-space-x-start: theme("padding.3");
+  }
 }
 
-:host(:focus) .container {
-  @apply text-color-1 no-underline;
+// Will always be .dropdown-link (<a>) or .dropdown-item-content(<span><slot></span>). Never both.
+.dropdown-link {
+  @apply focus-base;
 }
-
-:host([selected]) .container:not(.container--none-selection),
-:host([selected]) .container--link .dropdown-link {
-  @apply text-color-1 font-medium;
-  & calcite-icon {
-    color: theme("backgroundColor.brand");
-  }
+.dropdown-item-content {
+  @apply flex-auto py-0.5;
+  padding-inline-end: theme("margin.auto");
+  padding-inline-start: theme("margin.1");
 }
 
 // item icon
-.dropdown-item-icon {
+// Selection Indicator Icon
+calcite-icon.dropdown-item-icon {
+  --calcite-icon-color: var(
+    --calcite-dropdown-item-indicator-color,
```

Comment:
Sidebar: I think I've been implementing this pattern a bit differently (the public prop is resolved first when assigning internal CSS props). Will review and update accordingly.